editor/src/clj/editor/gui.clj

```diff
@@ -774,17 +786,27 @@
   (property y-anchor g/Keyword (default :yanchor-none)
             (dynamic edit-type (g/constantly (properties/->pb-choicebox Gui$NodeDesc$YAnchor))))
 
+  (property material g/Str
+            (default "")
+            (dynamic edit-type (g/fnk [material-names] (optional-gui-resource-choicebox material-names)))
+            (dynamic error (g/fnk [_node-id material-names material] (validate-material _node-id material-names material))))
+
   (output visual-base-node-msg g/Any produce-visual-base-node-msg)
   (output gui-scene g/Any (g/fnk [_node-id]
                                  (node->gui-scene _node-id)))
 
+  (output override-material-shader ShaderLifecycle (g/fnk [material-shaders material]
```

Comment:
We should probably rename this output, the GuiNode already has a :material-shader so I don't think it can be called material-shader..

Reply:
It looks like the old `material-shader` inputs and outputs were simply used to pass along the shader of the default material selected for the `GuiSceneNode`. We could get rid of it and instead pass that in as the "no material" entry associated with the empty string key in the `material-shaders` inputs & outputs you've introduced in this PR. That's very similar to how we do it for textures, and you already have code that falls back to the `""` entry in the `material-shaders` map elsewhere in this PR since you based it on how we do it for textures.